python/feldera/sql_context.py

```diff
@@ -0,0 +1,278 @@
+import time
+import pandas
+import re
+
+from typing import Optional, Dict
+from typing_extensions import Self
+from queue import Queue
+
+from feldera import FelderaClient
+from feldera.rest.program import Program
+from feldera.rest.pipeline import Pipeline
+from feldera._sql_table import SQLTable
+from feldera.sql_schema import SQLSchema
+from feldera.output_handler import OutputHandler
+from feldera.output_handler import _OutputHandlerInstruction
+from enum import Enum
+
+
+class BuildMode(Enum):
+    CREATE = 1
+    GET = 2
+    GET_OR_CREATE = 3
+
+
+def _table_name_from_sql(ddl: str) -> str:
+    return re.findall(r"[\w']+", ddl)[2]
+
+
+class SQLContext:
+    """
+    The SQLContext is the main entry point for the Feldera SQL API.
+    Abstracts the interaction with the Feldera API and provides a high-level interface for SQL pipelines.
+    """
+    client: FelderaClient
+    pipeline_name: str
+    program_name: str
+    build_mode: BuildMode
+
+    pipeline_description: str = ""
+    program_description: str = ""
+    ddl: str = ""
+    views: Dict[str, str] = {}
```

Comment:
The ordering of declarations matters in SQL, so these have to be arrays, not dictionaries